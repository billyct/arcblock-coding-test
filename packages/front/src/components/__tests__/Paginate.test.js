import {render} from '@testing-library/react'

jest.mock('../../hooks/useBlock')
const useBlock = require('../../hooks/useBlock')

jest.mock('../../hooks/useBlockRouteParams')
const useBlockRouteParams = require('../../hooks/useBlockRouteParams')

jest.mock('../PrevButton', () => 'prev-button-component')
jest.mock('../NextButton', () => 'next-button-component')

const Paginate = require('../Paginate').default

describe.each([
  [
    {block:{transactionsCount: 8}},
    {page: 10},
  ],
  [
    {block:{transactionsCount: 2000}},
    {page: 10},
  ]
])('test Paginate Component with useBlock(%o)', (block, blockRouteParams) => {
  beforeAll(() => {
    useBlock.__setMockData(block)
    useBlockRouteParams.__setMockData(blockRouteParams)
  })

  it('should render correct', function () {
    const {container} = render(<Paginate/>)
    expect(container).toMatchSnapshot()
  })
})
