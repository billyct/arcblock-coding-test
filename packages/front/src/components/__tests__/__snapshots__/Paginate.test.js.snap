// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Paginate Component with useBlock({ block: { transactionsCount: 8 } }) should render correct 1`] = `<div />`;

exports[`test Paginate Component with useBlock({ block: { transactionsCount: 2000 } }) should render correct 1`] = `
<div>
  <div
    class="flex justify-between p-6 pb-0 border-t border-gray-200"
  >
    <div>
      Showing 
      91
       to 
      100
       of 
      2000
       results
    </div>
    <div
      class="flex justify-end items-center"
    >
      <prev-button-component />
      <div
        class="border-r-2 border-gray-200 mx-2 inline-block h-4 "
      />
      <next-button-component />
    </div>
  </div>
</div>
`;
